app/scripts/controllers/alert-controller.test.ts

```diff
@@ -28,230 +28,196 @@ const EMPTY_ACCOUNT = {
     importTime: 0,
   },
 };
-describe('AlertController', () => {
-  let controllerMessenger: ControllerMessenger<
-    AlertControllerActions | AllowedActions,
-    | AlertControllerEvents
-    | KeyringControllerStateChangeEvent
-    | SnapControllerStateChangeEvent
-    | AllowedEvents
+
+type WithControllerOptions = Partial<AlertControllerOptions>;
+
+type WithControllerCallback<ReturnValue> = ({
+  controller,
+}: {
+  controller: AlertController;
+  messenger: ControllerMessenger<
+    AllowedActions | AlertControllerGetStateAction,
+    AllowedEvents | AlertControllerStateChangeEvent
   >;
-  let alertController: AlertController;
+}) => ReturnValue;
+
+type WithControllerArgs<ReturnValue> =
+  | [WithControllerCallback<ReturnValue>]
+  | [WithControllerOptions, WithControllerCallback<ReturnValue>];
 
-  beforeEach(() => {
-    controllerMessenger = new ControllerMessenger<
-      AllowedActions,
-      AllowedEvents
-    >();
-    controllerMessenger.registerActionHandler(
-      'AccountsController:getSelectedAccount',
-      () => EMPTY_ACCOUNT,
-    );
+async function withController<ReturnValue>(
+  ...args: WithControllerArgs<ReturnValue>
+): Promise<ReturnValue> {
+  const [{ ...rest }, fn] = args.length === 2 ? args : [{}, args[0]];
+  const { ...alertControllerOptions } = rest;
 
-    const alertMessenger = controllerMessenger.getRestricted({
+  const controllerMessenger = new ControllerMessenger<
+    AllowedActions | AlertControllerGetStateAction,
+    AllowedEvents | AlertControllerStateChangeEvent
+  >();
+
+  const alertControllerMessenger: AlertControllerMessenger =
+    controllerMessenger.getRestricted({
       name: 'AlertController',
-      allowedActions: [`AccountsController:getSelectedAccount`],
-      allowedEvents: [`AccountsController:selectedAccountChange`],
+      allowedActions: ['AccountsController:getSelectedAccount'],
+      allowedEvents: ['AccountsController:selectedAccountChange'],
     });
 
-    alertController = new AlertController({
-      state: {
-        unconnectedAccountAlertShownOrigins: {
-          testUnconnectedOrigin: false,
-        },
-        web3ShimUsageOrigins: {
-          testWeb3ShimUsageOrigin: 0,
-        },
-      },
-      controllerMessenger: alertMessenger,
-    });
+  controllerMessenger.registerActionHandler(
+    'AccountsController:getSelectedAccount',
+    jest.fn().mockReturnValue(EMPTY_ACCOUNT),
+  );
+
+  const controller = new AlertController({
+    messenger: alertControllerMessenger,
+    state: getDefaultAlertControllerState(),
```

Comment:
Going along with a previous comment, if `state` is optional then we shouldn't need to mix in the default state as the constructor already does that:

``` suggestion
```